### DISCUSSION TITLE
Communication with the API failed, is NPM running correctly?

### DISCUSSION THREAD

**Discussion by tgaertner** [NginxProxyManager/nginx-proxy-manager#2528] [Q&A]
I get this Error message on trying  to "Test Server Reachability" on  "Add Let's Encrypt Certificate" dialog.

The domain point to my IP. The nginx-proxy-manager is forwarding port 80 already successfull to my service.
Anyone an idea what I'm missing here?

**Comment by punkyard** [comment]
hi
same for me here.
Domain points to IP server/nginx - through cloudflare with proxy disabled.

**Comment by tgaertner** [comment]
Hint: I found out, that just the test fails. Its still possible to add certificates.

**Reply by punkyard** [reply]
yes, but why does the test fails? ...
Is it because of cloudflare or else, or the certificate module is failing? 

Personnaly, I can't add custom certificates from cloudflare - yet they do work on another server running hestia control panel, which uses nginx. I had to add cloudflare to certbot.

So I tried to add cloudflare to my nginx-pm container, but it fails ..
`python3 -m pip install --upgrade pip
pip install --upgrade --force-reinstall acme==2.2.0
pip install certbot-dns-cloudflare==$(certbot --version | grep -Eo '0-9+') cloudflare`

Here is my contribution ✌️🤞

**Reply by megabert** [reply]
Same here. NPM is behind a HAproxy (pfSense) here. Test is failing but certificates are being registered correctly.
